drivers/soundwire/bus.c

```diff
@@ -877,7 +874,7 @@ static int sdw_slave_clk_stop_prepare(struct sdw_slave *slave,
 			val |= SDW_SCP_SYSTEMCTRL_WAKE_UP_EN;
 	} else {
 		ret = sdw_read_no_pm(slave, SDW_SCP_SYSTEMCTRL);
-		if (ret < 0) {
+		if (ret < 0 && ret != -ENODATA) {
 			dev_err(&slave->dev, "SDW_SCP_SYSTEMCTRL read failed:%d\n", ret);
 			return ret;
 		}
```

Comment:
ok, this is now wrong: if `ret == -ENODATA` the `if` won't be entered. So the function will proceed below to assign `val = ret & ~SDW_SCP_SYSTEMCTRL_CLK_STP_PREP` (btw, can we modify those two lines below to this simpler form?) and write it to `SDW_SCP_SYSTEMCTRL`.